Source/WebKit/UIProcess/Cocoa/WKWebView+TextExtraction.swift

```diff
@@ -23,13 +23,12 @@
 
 #if canImport(UIIntelligenceSupport)
 
-@_spiOnly import WebKit
-@_spiOnly import UIIntelligenceSupport
+@_weakLinked @_spiOnly import UIIntelligenceSupport
 
 #if canImport(UIKit)
-@_spi(UIIntelligenceSupport) import UIKit
+@_weakLinked @_spi(UIIntelligenceSupport) import UIKit
 #else
-@_spi(UIIntelligenceSupport) import AppKit
+@_weakLinked @_spi(UIIntelligenceSupport) import AppKit
```

Comment:
I would expect to see some `#_hasSymbol` checks accompanying these. Is there a reason we don't need them?

In general every symbol linked against has to be marked weak for dyld to actually allow loading of the binary to fail. I'm not sure what Swift does when you @_weakLink something without checking for symbol availability, but it might do the safe thing of marking the use of the symbol as strong binding, which would lead to it not being weak-linked at all.

Reply:
wait actually why do UIKit and AppKit themselves need to be weak linked?

Reply:
> but it might do the safe thing of marking the use of the symbol as strong binding

Oof, that's unfortunate. I'll audit these call sites, good catch!

Reply:
I've been convinced, r=me

Reply:
> wait actually why do UIKit and AppKit themselves need to be weak linked?

You're right — those 2 `@_weakLinked` shouldn't be necessary. I removed them in the latest version.